macros/contexts/contextSignificantFigures.pl

```diff
@@ -136,72 +131,62 @@ sub new {
 # Either return the current number of sigfigs for the number or set the current number.
 
 sub sigfigs {
-	my ($self, $num_sigfigs) = @_;
-	return $self->{sigfigs} unless defined($num_sigfigs);
-	Value::Error('The number of significant figures must be inf or an integer >=-1')
-		unless $num_sigfigs =~ m/^inf$/ || ($num_sigfigs =~ /-?\d+/ && $num_sigfigs >= 1);
-	my ($v, $exp);
-	if ($num_sigfigs =~ m/^inf$/) {
-		$v   = $self->value;
-		$exp = 0;
-	} else {
-		($v, $exp) = split(/E/, $self->value);
-		$exp = 0 unless defined($exp);    # if $self->value is not in exponential form.
-	}
-
-	# Calculate the digit to round to.
-	if ($num_sigfigs !~ /^inf$/) {
-		my $N = $num_sigfigs - 1;
-		$self->{data} = [ sprintf("%.${N}E", $v * 10**$exp) ];
-	} else {
-		$self->{data} = [$v];
-	}
-
-	$self->{sigfigs} = $num_sigfigs;
+	my ($self, $n) = @_;
+	return $self->{sigfigs} if !defined($n) || $self->{sigfigs} == $n;
+	$self->{sigfigs} = $self->context->checkSigFigs($n);
+	$self->{data}[0] = $n == 'inf' ? $self->value : $self->format('E', ROUND($self->value, $n), $n);
```

Comment:
When you stop storing the string value and store the perl one instead, this should be

``` perl
	$self->{data}[0] = ROUND($self->value, $n);
```